### PR TITLE
created orderBy function

### PR DESCRIPTION
**Pull request by laravelartisan**

### CODE REVIEW

File: .idea/.name [hunk 1]
```diff
@@ -0,0 +1 @@
+30-seconds-of-php-code
```
Comment (peter279k) [inline .idea/.name]: Do you follow the [CONTRIBUTING](https://github.com/appzcoder/30-seconds-of-php-code/blob/master/CONTRIBUTING.md) guide?
This file is not necessary.

File: .idea/30-seconds-of-php-code.iml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+          <root url="phar://$MODULE_DIR$/vendor/phar-io/manifest/tests/_fixture/test.phar" />
+        </CLASSES>
+        <SOURCES>
+          <root url="phar://$MODULE_DIR$/vendor/phpunit/phpunit/tests/_files/phpunit-example-extension/tools/phpunit.d/phpunit-example-extension-1.0.1.phar" />
+          <root url="phar://$MODULE_DIR$/vendor/phar-io/manifest/tests/_fixture/test.phar" />
+        </SOURCES>
+      </library>
+    </orderEntry>
+  </component>
+</module>
```
Comment (peter279k) [inline .idea/30-seconds-of-php-code.iml]: This file is not necessary.
You have to follow the [CONTRIBUTING](https://github.com/appzcoder/30-seconds-of-php-code/blob/master/CONTRIBUTING.md) guide to do the contribution.

File: .idea/misc.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <OptionsSetting value="true" id="Remove" />
+    <OptionsSetting value="true" id="Checkout" />
+    <OptionsSetting value="true" id="Update" />
+    <OptionsSetting value="true" id="Status" />
+    <OptionsSetting value="true" id="Edit" />
+    <ConfirmationsSetting value="0" id="Add" />
+    <ConfirmationsSetting value="0" id="Remove" />
+  </component>
+  <component name="ProjectRootManager" version="2" />
+</project>
```
Comment (peter279k) [inline .idea/misc.xml]: This file is not necessary.
You have to follow the CONTRIBUTING guide to do the contribution.

File: .idea/modules.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/30-seconds-of-php-code.iml" filepath="$PROJECT_DIR$/.idea/30-seconds-of-php-code.iml" />
+    </modules>
+  </component>
+</project>
```
Comment (peter279k) [inline .idea/modules.xml]: This file is not necessary.
You have to follow the CONTRIBUTING guide to do the contribution.

File: .idea/vcs.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (peter279k) [inline .idea/vcs.xml]: This file is not necessary.
You have to follow the CONTRIBUTING guide to do the contribution.

File: .idea/workspace.xml [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    <entry file="file://$PROJECT_DIR$/tests/ArrayTest.php">
+      <provider selected="true" editor-type-id="text-editor">
+        <state vertical-scroll-proportion="-1.0651408">
+          <caret line="234" column="33" selection-start-line="234" selection-start-column="33" selection-end-line="234" selection-end-column="33" />
+          <folding />
+        </state>
+      </provider>
+    </entry>
+  </component>
+</project>
```
Comment (peter279k) [inline .idea/workspace.xml]: This file is not necessary.
You have to follow the CONTRIBUTING guide to do the contribution.

File: src/helpers.php [hunk 7]
```diff
@@ -304,3 +304,47 @@ function clampNumber($num, $a, $b)
 {
     return max(min($num, max($a, $b)), min($a, $b));
 }
+
+function orderBy(array $items, $sortingAttr, $sortingType = 'asc')
+{
+    if(is_array($items) && !empty($items)){
```
Comment (peter279k) [inline src/helpers.php]: `!empty` is not good here...
How about using `is_array` directly?

File: src/helpers.php [hunk 8]
```diff
@@ -304,3 +304,47 @@ function clampNumber($num, $a, $b)
...
+        }
+
+        foreach($sortedItems as $sortedItemVals){
+            foreach($sortedItemVals as $sortedItemVal){
+                $sortedItemCollection [] = $sortedItemVal;
+            }
+        }
+        return $sortedItemCollection;
+    }
+}
```
Comment (peter279k) [inline src/helpers.php]: The file should follow the PSR2 coding style.

**Comment by peter279k** [comment]
It looks good to me. @sohelamin, please take a look at that at available time.
But it looks like the commits is very large. Please squash all the commits to one commit if possible.

**Comment by sohelamin** [comment]
Thank you both of you guys 👍 
Actually this project's main goal is to make simple code snippets so people can understand it within just 30 seconds.
This seems a long method. @laravelartisan I appreciate your effort but please if you can make it simple and small or I will refactor it when I get some times :)

**Comment by laravelartisan** [comment]
Thanks to both of you @sohelamin and @peter279k  for your feedback and instructions. @sohelamin I tried to make the function small and simple. Please, check it out when possible. Waiting for your feedback.

**Comment by sohelamin** [comment]
Thanks a lot @laravelartisan 
I will merge and refactor a bit :)

**Pull request merged**